hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -590,6 +590,8 @@ private void doCommit(String instant, List<WriteStatus> writeResults) {
         reset();
         this.ckpMetadata.commitInstant(instant);
         LOG.info("Commit instant [{}] success!", instant);
+        this.flinkTimeLineMetrics.updateTimeLineMetrics(this.writeClient.getHoodieTable().getActiveTimeline());
```

Comment:
The `BaseHoodieClient` already got a `HoodieMetrics` instance, we can just enhance it with new items or add the `FlinkTimeLineMetrics` in `HoodieFlinkWriteClient`, and update it in `HoodieFlinkWriteClient.postCommit`, you need to override the `#postCommit` method so that we can eliminate the redundant timeline loading(file listing) for each commit.

Reply:
Thanks for advising, finally enhanced in `HoodieMetrics `.   @danny0405